### ISSUE TITLE
When kubelet is restarted, the endpoints of the service are lost temporarily

### ISSUE THREAD

**Issue by zhaizeyu** [kubernetes/kubernetes#127085]
### What happened?

My k8s version is 1.22.1,
service type is Cluster.
When i restart kubelet on business node and use this cmd on k8s node:kubectl get endpoints fkft7-nslb-north-svc -nns000000000000000000001，i find the endpoints of the service are lost temporarily. and log of kube-controller-manager show me like this
![image](https://github.com/user-attachments/assets/62659cbf-5aae-4b3d-ba65-08c00fa9781f)

Are other conditions triggering this bug as well? I wonder if this problem has been fixed.

### What did you expect to happen?

endpoints won't lost

### How can we reproduce it (as minimally and precisely as possible)?

restart kubelet and watch endpoints of service

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>
1.22.1

### Cloud provider

<details>

</details>
None

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>
suse 12.5

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by tamilselvan1102** [comment]
@zhaizeyu 
k8s version 1.22.1 is older version. There is not support provided for it. 
Please update to latest version 1.31 and confirm the same.

**Comment by neolit123** [comment]
/sig network

> Please update to latest version 1.31 and confirm the same.

+1

**Comment by KartikeyBartwal** [comment]
Where is your kubernetes running ? Is it AWS EKS or are your working on some other or local environment ?

As of now, these are the **latest versions of kubernetes** **(1.31, 1.30, 1.29)**. 

Since kubernetes brings updates every 3-4 months, it is advised to **stick to at least one of the top 3 versions** for your tasks, or possibly regularly upgrading and migrating your environment. AWS makes it compulsory for their users to upgrade their EKS version after every 1 year, for stability and security reasons

**Comment by zhaizeyu** [comment]
i found the issue fixed after 1.25.x
